google/cloud/internal/opentelemetry.cc

```diff
@@ -0,0 +1,32 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
```

Comment:
Why the `#ifdef` guards? There is nothing here that depends on OpenTelemetry being linked or available?

Reply:
That is true, but when we add types that do depend on OpenTelemetry, I do not want to be micromanaging the `#ifdef` guards.

Reply:
Hmmm... Okay.